### ISSUE TITLE
Upgrade Anim04

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/gmt#5778]
I am upgrading anim04. Below is the animation based on the earth_relief grid. I should try a higher resolution but I had some issues (that I think it will be solved with Paul PR). 

I have two doubts/questions:

1. I think it would be better to use a satellite image. Maybe the black Marble (see the image below). I think we could use it a bit more. 
2. I don't understand well the General Prospective projection. In the black marble image, I reduce it and I see that I get a rectangular image. In the animation the lower part was outside the canvas. Is it ok that way (like in the script)? Or what parameters should I use for -JG?

https://user-images.githubusercontent.com/17908951/133854154-881ec849-72d8-4d74-9822-4e195b661f47.mp4

![anim04_sat](https://user-images.githubusercontent.com/17908951/133855084-1297715a-b827-486a-aef2-73acd4ecd492.png)

Full script:

```
if [ $# -eq 1 ]; then	# Just make master PostScript frame 27 (az = 206)
	opt="-M27,png -Fmp4"
else	# Make animated GIF, infinitely looping
	opt="-M27,png"
fi

# 1. Create files needed in the loop
cat << 'EOF' > pre.sh
gmt begin
	gmt project -C-73.8333/40.75 -E-80.133/25.75 -G10 -Q > flight_path.txt
	gmt grdcut @earth_relief_30s -R-100/-70/10/50 -GUSEast_Coast.nc
	gmt grdgradient USEast_Coast.nc -A90 -Nt1 -Gint_US.nc
	gmt makecpt -Cglobe -H > globe_US.cpt
gmt end
EOF
# 2. Set up the main frame script
cat << 'EOF' > main.sh
gmt begin
	gmt grdimage -JG${MOVIE_COL0}/${MOVIE_COL1}/160/210/55/0/36/34/${MOVIE_WIDTH}+du -Rg USEast_Coast.nc -Iint_US.nc -Cglobe_US.cpt -Xc -Y-3c
	gmt plot -W1p flight_path.txt
gmt end
EOF
# 3. Run the movie
gmt movie main.sh -Chd -Nanim04 -Tflight_path.txt -Sbpre.sh -Zs -H2 -Lf+o0.25c+f14p,Helvetica-Bold $opt
```

**Comment by PaulWessel** [comment]
Thanks @Esteban82.  I am working on a PR to replace the horrible -JG syntax with

`-Jg|G<lon>/<lat>/scale|width>[+a<azimuth>][+t<tilt>][+v<width>[/<height>]][+w<twist>][+z<altitude>[r]]`

It should make our -JG examples much simpler to understand.  I will also make some documentation to explain the settings.  Perhaps waiting a bit with anim04 will be helpful so I can finish this - hopefully today?

**Comment by Esteban82** [comment]
Ok. Perfect.

**Comment by Esteban82** [comment]
I make this animation (with a 02m resolution). I got a an error with higher resolution. I think that the satellite image would be more appropriate for this animation (if the higher resolution works) since the descriptions says: " simulates what an imaginary satellite might see as it passes in a great circle from New York to Miami".

https://user-images.githubusercontent.com/17908951/134012782-b438b535-e167-48a0-a80d-3d29cb291bfe.mp4

I try a higher resolution but I got these errors (when a try to make the animation, not with a single master frame): 

```
movie_frame.sh: línea 14: 54689 Terminado (killed)      gmt grdimage -JG${MOVIE_COL0}/${MOVIE_COL1}/${MOVIE_WIDTH}+du+z160+a210+t55+v36 -Rg @earth_night_01m -Xc -Y-5c
movie_frame.sh: línea 14: 54687 Terminado (killed)      gmt grdimage -JG${MOVIE_COL0}/${MOVIE_COL1}/${MOVIE_WIDTH}+du+z160+a210+t55+v36 -Rg @earth_night_01m -Xc -Y-5c
movie_frame.sh: línea 14: 54688 Terminado (killed)      gmt grdimage -JG${MOVIE_COL0}/${MOVIE_COL1}/${MOVIE_WIDTH}+du+z160+a210+t55+v36 -Rg @earth_night_01m -Xc -Y-5c
movie_frame.sh: línea 14: 54691 Terminado (killed)      gmt grdimage -JG${MOVIE_COL0}/${MOVIE_COL1}/${MOVIE_WIDTH}+du+z160+a210+t55+v36 -Rg @earth_night_01m -Xc -Y-5c
end [WARNING]: Figure # 1 (../anim04_sat2C_001) was registered but no matching PostScript-|+ file found - skipping
Error: /undefined in PSL_movie_label_completion
Operand stack:
   PSL_eoc
Execution stack:
   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push   1990   1   3   %oparray_pop   1989   1   3   %oparray_pop   1977   1   3   %oparray_pop   1833   1   3   %oparray_pop   --nostringval--   %errorexec_pop   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--
Dictionary stack:
   --dict:747/1123(ro)(G)--   --dict:0/20(G)--   --dict:75/200(L)--   --dict:147/250(L)--
Current allocation mode is local
Current file position is 20459
GPL Ghostscript 9.50: Unrecoverable error, exit code 1

```

```
if [ $# -eq 0 ]; then	# Just make master PostScript frame 27 (az = 206)
	opt="-M27,png -Fmp4"
else	# Make animated GIF, infinitely looping
	opt="-M27,png"
fi

# 1. Create files needed in the loop
cat << 'EOF' > pre.sh
gmt begin
	gmt project -C-73.8333/40.75 -E-80.133/25.75 -G10 -Q > flight_path.txt
gmt end
EOF
# 2. Set up the main frame script
cat << 'EOF' > main.sh
gmt begin
	gmt grdimage -JG${MOVIE_COL0}/${MOVIE_COL1}/${MOVIE_WIDTH}+du+z160+a210+t55+v36 -Rg  @earth_night_02m -Xc -Y-5c #  ${MOVIE_WIDTH}+du
#	gmt plot -W1p flight_path.txt
gmt end
EOF
# 3. Run the movie
gmt movie main.sh -Chd -Nanim04_sat2C -Tflight_path.txt -Sbpre.sh -Zs -H2 -Lf+o0.25c+f14p,Helvetica-Bold $opt


```

**Comment by PaulWessel** [comment]
I will try to reproduce the problems here with 01m.

**Comment by PaulWessel** [comment]
Worked fine with 01m resolution for me. How many cores does the movie use for you, and how much RAM might that be per core? I.e., I wonder if you run out of memory.  Try -x3 to run only movie plus two animations at the time to see if it works then, then change to 5, 7 etc.

**Comment by Esteban82** [comment]
Ok. My pc has 8 cores and 8 Gb of ram. I try with -x5 and failed. Now it is running with -x3 it is working so far.

By the way, How -x works? When I put -x3 I get the message `movie [INFORMATION]: Build frames using 2 cores`. Is it really 3 cores? Does it start counting from 0?

**Comment by PaulWessel** [comment]
I leave one core for the movie process and then use the rest for the frames.  So 3 is 3 but it is 1 + 2.

**Comment by Esteban82** [comment]
Mmm I wonder if it would be better is the information says 3 cores.

**Comment by PaulWessel** [comment]
Yes, this could be clarified.  At least the movie rst sound say so.  I just added that to movie and batch usage and docs.

**Comment by Esteban82** [comment]
It work. This is the animation with 01m resolution. I think it looks nice. Maybe the 30s would be better but it would take a lot of time to make on my pc. 


https://user-images.githubusercontent.com/17908951/134098120-495bf69d-8d8b-480a-897a-726f70b6f5c7.mp4

**Comment by PaulWessel** [comment]
That is really nice.  Yes, it might look even better at 30s.  I can give it a test to see if it is worth it.  Why dont you make a pull request for the updated anim and then I can use that branch to edit and rn with 30s and see. If it is worth it I can suggest that change and I can upload the 30s mp4.

**Comment by PaulWessel** [comment]
Wonder if something other than white would be better for the horizon? IT is night time so white is odd, no? Maybe dark blue, like navy?

**Comment by Esteban82** [comment]
Right. Midnightblue looks better. I try to use a [pattern](https://docs.generic-mapping-tools.org/latest/_images/GMT_App_E.png) to simulate the stars (like 30 or 31) but it didn't work. 

![anim04](https://user-images.githubusercontent.com/17908951/134100842-0517169b-12f3-47ad-bb37-0e3240a50e19.png)

**Comment by PaulWessel** [comment]
I am not sure (yet) how to do this, but if we could make a grid with elevation above the horizon we could lay down a grdimage with a cpt to simulate the night sky going form lighter to darker away from the horizon.

**Comment by PaulWessel** [comment]
OK, the problem here is that grdimage cannot cut out the required image subset - it reads the whole thing.  So at 01m or (worse) 30s that is a lot of memory (3.5 Gb for the 30s image).  And movie runs those in parallel, so even my 64Gb memory soon evaporates.  @joa-quim could not get this fixed so I think I need to take a deep-dive on this problem since we cannot go on living with this limitation.  That is why -x needs to be used if we push the resolution up.

**Comment by Esteban82** [comment]
So, Paul, should I wait until you find solve the limitation, right?

**Comment by PaulWessel** [comment]
Yes, I think so.  It is just too slow as it is.

**Comment by meghanrjones** [comment]
The current version of animation 4 looks broken: https://docs.generic-mapping-tools.org/dev/animations/anim04.html#anim04, possibly due to the -JG updates. We may have some issues with broken backwards compatibility.
